### PR TITLE
feat: add 'featured' tagType to Tag

### PR DESCRIPTION
**Pull request by nikazawila**
Adding `featured` tagType to Tag, as requested by @fabe 

Screenshot 2021-03-24 at 13 35 51

### CODE REVIEW

**Comment by mshaaban0** [comment]
Should this go to the badge component a.k.a tag v4?

**Review by domarku — APPROVED** [review]

File: packages/forma-36-react-components/src/components/Tag/Tag.tsx [hunk 1]
```diff
@@ -32,7 +33,7 @@ const statusTagTypeMap = {
 export interface TagProps {
   /**
    * prop, used for setting type of the component from
-   * TagTypes - primary, primary-filled, positive, negative, warning, secondary, muted */
+   * TagTypes - primary, primary-filled, positive, negative, warning, secondary, muted, featured */
```
Comment (denkristoffer) [inline packages/forma-36-react-components/src/components/Tag/Tag.tsx:36]: Maybe we can just remove this list in the comment, it should show the options in your editor anyway and we don't really do this anywhere else.

**Review by denkristoffer — APPROVED** [review]
Not related to this PR itself by why does the CHANGELOG files keep getting changed and show up in every PR? It's like they go back and forth between formatting. Maybe our prettier setup is wrong...

**Review by burakukula — APPROVED** [review]
Looks good! 🙌
Maybe we can also add a comment about this new variant in `Tag` README file? WDYT?
+1 for the comment from @mshaaban0 -> `Badge`  component should also contain this variant (separate PR that can be merged into v4 branch I guess?).

**Review by fabe — APPROVED** [review]
LGTM, do we also add this to storybook?

**Comment by nikazawila** [comment]
@denkristoffer I also found those linting failures very mysterious - not really sure what's the answer to that.

**Review by gui-santos — APPROVED** [review]

**Pull request merged**